Examples/Physics_applications/laser_ion/analysis_test_laser_ion.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+
+import numpy as np
+import openpmd_api as io
+
+sys.path.insert(1, "../../../../warpx/Regression/Checksum/")
+from checksumAPI import evaluate_checksum
+
+
+def load_field_from_iteration(
+    series, iteration: int, field: str, coord: str = None
+) -> np.ndarray:
+    """Load iteration of field data from file."""
+
+    it = series.iterations[iteration]
+    field_obj = it.meshes[f"{field}"]
+
+    if field_obj.scalar:
+        field_data = field_obj[io.Mesh_Record_Component.SCALAR].load_chunk()
+    elif coord in [item[0] for item in list(field_obj.items())]:
+        field_data = field_obj[coord].load_chunk()
+    else:
+        raise Exception(
+            f"Specified coordinate: f{coord} is not available for field: f{field}."
+        )
+    series.flush()
+
+    return field_data
+
+
+def compare_time_avg_with_instantaneous_diags(dir_inst: str, dir_avg: str):
+    """Compare instantaneous data (multiple iterations averaged in post-processing) with in-situ averaged data."""
+
+    field = "E"
+    coord = "z"
+    avg_period_steps = 5
+    avg_output_step = 100
+
+    path_tpl_inst = f"{dir_inst}/openpmd_%T.h5"
+    path_tpl_avg = f"{dir_avg}/openpmd_%T.h5"
+
+    si = io.Series(path_tpl_inst, io.Access.read_only)
+    sa = io.Series(path_tpl_avg, io.Access.read_only)
+
+    ii0 = si.iterations[0]
+    fi0 = ii0.meshes[field][coord]
+    shape = fi0.shape
+
+    data_inst = np.zeros(shape)
+
+    for i in np.arange(avg_output_step - avg_period_steps + 1, avg_output_step + 1):
+        data_inst += load_field_from_iteration(si, i, field, coord)
+
+    data_inst = data_inst / avg_period_steps
+
+    data_avg = load_field_from_iteration(sa, avg_output_step, field, coord)
+
+    # Compare the data
+    if np.allclose(data_inst, data_avg):
```

Comment:
I think the default tolerances for NumPy's `allclose` are `rtol=1e-05` and `atol=1e-08` (seen at https://numpy.org/doc/stable/reference/generated/numpy.allclose.html). Shall we use stricter tolerances for this comparison?

Reply:
Uh, I don't know. I think they are pretty strict already because the absolute values are on the order of `1e8`.

Reply:
I see, I think I'm mostly concernced about the relative tolerance, where the order of the absolute values in the calculation of things like $|x_{avg}-x_{inst}|/|x_{inst}|$ should not play a role as it's factored out.

Reply:
Right, that's fair. But what should we set it to?
Weren't some other recent tests failing because of too strict tolerance limits?

Reply:
We definitely had that issue with checksums, but I thought this could be more controlled. Sometimes we do expect close results, for example when we check restarts.

I guess I would be interested in making the tolerance here as strict as possible, just for the sake of making sure that we don't overlook future bugs.

I think `np.allclose` must be used with care. They have some info in the docs, e.g., about the fact that `atol` must be set with care otherwise the function won't work for small numbers. For instance, `np.allclose(a=1e-9, b=2e-9)` returns `True`, which one probably would not want, because of the default values of `rtol` and `atol` (in this case the issue is with `atol`).

In our case I think the default `atol` is okay because the order of magnitude of the values is $10^8$, but we can maybe try to lower `rtol` as much as possible and see if it causes instabilities when we start running this test regularly.

Reply:
```suggestion
    if np.allclose(data_inst, data_avg, rtol=1e-12):
```